From 13e65a0c470b25bdd70e632738784d544d65d63a Mon Sep 17 00:00:00 2001
Message-Id: <13e65a0c470b25bdd70e632738784d544d65d63a.1368111913.git.minovotn@redhat.com>
In-Reply-To: <405603258af5154387bea676be1f904b6713f6ae.1368111913.git.minovotn@redhat.com>
References: <405603258af5154387bea676be1f904b6713f6ae.1368111913.git.minovotn@redhat.com>
From: Amit Shah <amit.shah@redhat.com>
Date: Wed, 24 Apr 2013 08:17:39 +0200
Subject: [PATCH 05/65] Revert "char: Throttle when host connection is down"

RH-Author: Amit Shah <amit.shah@redhat.com>
Message-id: <80cd0e463c98ca9cd35d470432b15edfea43e1e5.1366724981.git.amit.shah@redhat.com>
Patchwork-id: 50783
O-Subject: [RHEL6.5 qemu-kvm PATCH 05/65] Revert "char: Throttle when host connection is down"
Bugzilla: 909059
RH-Acked-by: Hans de Goede <hdegoede@redhat.com>
RH-Acked-by: Gerd Hoffmann <kraxel@redhat.com>
RH-Acked-by: Paolo Bonzini <pbonzini@redhat.com>

This reverts commit b1f45e402af97406d4dd2fb998f419d342b88384.

Signed-off-by: Amit Shah <amit.shah@redhat.com>
---
 qemu-char.c | 14 --------------
 1 file changed, 14 deletions(-)

Signed-off-by: Michal Novotny <minovotn@redhat.com>
---
 qemu-char.c | 14 --------------
 1 file changed, 14 deletions(-)

diff --git a/qemu-char.c b/qemu-char.c
index 1c086b3..d4c3311 100644
--- a/qemu-char.c
+++ b/qemu-char.c
@@ -141,9 +141,6 @@ static void qemu_chr_generic_open_bh(void *opaque)
 {
     CharDriverState *s = opaque;
     qemu_chr_event(s, CHR_EVENT_OPENED);
-    if (s->write_blocked) {
-        char_write_unblocked(s);
-    }
     qemu_bh_delete(s->bh);
     s->bh = NULL;
 }
@@ -2045,17 +2042,6 @@ static int tcp_chr_write(CharDriverState *chr, const uint8_t *buf, int len)
         ret = send_all(chr, s->fd, buf, len);
         if (ret == -1 && errno == EPIPE) {
             tcp_closed(chr);
-
-            if (chr->chr_enable_write_fd_handler && chr->chr_write_unblocked) {
-                /*
-                 * Since we haven't written out anything, let's say
-                 * we're throttled.  This will prevent any output from
-                 * the guest getting lost if host-side chardev goes
-                 * down.  Unthrottle when we re-connect.
-                 */
-                chr->write_blocked = true;
-                return 0;
-            }
         }
         return ret;
     } else {
-- 
1.7.11.7

